avalancheup-aws/src/apply/mod.rs

```diff
@@ -67,25 +117,418 @@ pub fn command() -> Command {
+        subnet::vm_name_to_id(&opts.chain_name)?
+    } else {
+        ids::Id::from_str(&opts.vm_id)?
+    };
+    log::info!("VM ID is {}", vm_id.to_string());
+
+    let resp = jsonrpc_client_info::get_network_id(&opts.chain_rpc_url)
+        .await
+        .unwrap();
+    let network_id = resp.result.unwrap().network_id;
+
+    let priv_key = key::secp256k1::private_key::Key::from_hex(&opts.key).unwrap();
+    let wallet_to_spend = wallet::Builder::new(&priv_key)
+        .base_http_url(opts.chain_rpc_url.clone())
+        .build()
+        .await
+        .unwrap();
+
+    let p_chain_balance = wallet_to_spend.p().balance().await.unwrap();
+    let p_chain_address = priv_key
+        .to_public_key()
+        .to_hrp_address(network_id, "P")
+        .unwrap();
+    log::info!(
+        "loaded wallet '{p_chain_address}', fetched its P-chain balance {} AVAX ({p_chain_balance} nAVAX, network id {network_id})",
+        units::cast_xp_navax_to_avax(primitive_types::U256::from(p_chain_balance))
     );
 
-    let mut spec =
-        avalanche_ops::aws::spec::Spec::load(spec_file_path).expect("failed to load spec");
-    spec.validate()?;
+    let mut all_node_ids = Vec::new();
+    let mut region_to_instance_ids: HashMap<String, Vec<String>> = HashMap::new();
+    for (node_id, region_machine_id) in target_nodes.iter() {
+        log::info!(
+            "will send SSM doc to the node '{node_id}' of '{}' in the region '{}'",
+            region_machine_id.machine_id,
+            region_machine_id.region,
+        );
+        all_node_ids.push(node_id.clone());
+
+        if let Some(instance_ids) = region_to_instance_ids.get_mut(&region_machine_id.region) {
+            instance_ids.push(region_machine_id.machine_id.clone());
+        } else {
+            region_to_instance_ids.insert(
+                region_machine_id.region.clone(),
+                vec![region_machine_id.machine_id.clone()],
+            );
+        }
+    }
+
+    // if all nodes need to be staked
+    println!();
+    let estimated_required_avax =
+        units::cast_avax_to_xp_navax(primitive_types::U256::from(opts.staking_amount_in_avax))
+            .checked_mul(primitive_types::U256::from(all_node_ids.len()))
+            .unwrap();
+    log::info!(
+        "required AVAX to validate all nodes {estimated_required_avax} nAVAX ({} AVAX)",
+        units::cast_xp_navax_to_avax(estimated_required_avax)
+    );
+    if primitive_types::U256::from(p_chain_balance) < estimated_required_avax {
+        log::warn!("'{p_chain_address}' only has {p_chain_balance}, not enough to validate all nodes (needs {estimated_required_avax} nAVAX)");
+        let selected = Select::with_theme(&ColorfulTheme::default())
+            .with_prompt("Should we still proceed?")
+            .items(&["Yes...?", "No!!!"])
+            .default(0)
+            .interact()
+            .unwrap();
+        if selected == 1 {
+            return Ok(());
+        }
+    }
+
```

Comment:
Can we add a flag to enable subnet installation? (and disable by default)?

Looks like this is run by default?